### ISSUE TITLE
**Question!**  The images sampled during the training process are different from those generated during inference.

### ISSUE THREAD

**Issue by zc1023** [Yuanshi9815/OminiControl#117]
Thank you for your interesting work. When I try to use your training script 'train_canny', the loss remains about 0.3 after 1k steps. However, the sampled images keep improving during training. 

Also, the image sampled during training in 4k steps is as follows.  

![Image](https://github.com/user-attachments/assets/48d0b465-9576-4725-a64b-d720bba59daf)

It's pretty good. But when using the weight trained after 4k steps, we generate such different result as follows.

![Image](https://github.com/user-attachments/assets/61b778a5-13f1-4c7f-b808-90607298a5a4)

Why is that?
Here is the key part of my inference code.

```python
pipe.load_lora_weights(
    "runs/20250412-183956/ckpt/4000/pytorch_lora_weights.safetensors",
    adapter_name = "canny"
)

image = Image.open(f"assets/vase_hq.jpg").convert("RGB")
condition = Condition("canny", image,)
w, h, min_dim = image.size + (min(image.size),)
    image = image.crop(
        ((w - min_dim) // 2, (h - min_dim) // 2, (w + min_dim) // 2, (h + min_dim) // 2)
    ).resize((512, 512))
prompt = "A beautiful vase on a table."
result_img = generate(
        pipe,
        prompt=prompt,
        conditions=[condition],
        model_config=model_config,
    ).images[0]
```

**Comment by Yuanshi9815** [comment]
Hi @zc1023 ,

You can try this:
```
pipe.load_lora_weights(
    "runs/20250412-183956/ckpt/4000/pytorch_lora_weights.safetensors",
    adapter_name = "canny"
)

image = Image.open(f"assets/vase_hq.jpg").convert("RGB")
w, h, min_dim = image.size + (min(image.size),)
image = image.crop(
    ((w - min_dim) // 2, (h - min_dim) // 2, (w + min_dim) // 2, (h + min_dim) // 2)
).resize((512, 512))


condition = Condition("canny", image,)

prompt = "A beautiful vase on a table."
result_img = generate(
        pipe,
        prompt=prompt,
        conditions=[condition],
        model_config=model_config,
    ).images[0]
```

I wish it works.

;D

**Comment by zc1023** [comment]
> Hi [@zc1023](https://github.com/zc1023) ,
> 
> You can try this:
> 
> ```
> pipe.load_lora_weights(
>     "runs/20250412-183956/ckpt/4000/pytorch_lora_weights.safetensors",
>     adapter_name = "canny"
> )
> 
> image = Image.open(f"assets/vase_hq.jpg").convert("RGB")
> w, h, min_dim = image.size + (min(image.size),)
> image = image.crop(
>     ((w - min_dim) // 2, (h - min_dim) // 2, (w + min_dim) // 2, (h + min_dim) // 2)
> ).resize((512, 512))
> 
> 
> condition = Condition("canny", image,)
> 
> prompt = "A beautiful vase on a table."
> result_img = generate(
>         pipe,
>         prompt=prompt,
>         conditions=[condition],
>         model_config=model_config,
>     ).images[0]
> ```
> 
> ;D

@Yuanshi9815 . Doesn't work.

**Comment by Yuanshi9815** [comment]
Hi @zc1023 , 

May I check the `model_config` in your code?

**Comment by zc1023** [comment]
Hi @Yuanshi9815 
```
    model_config = {
        "union_cond_attn": True,
        "add_cond_attn": False,
        "latent_lora": False,}
```
The `model_cofig` is set to the same as the training script.

**Comment by Yuanshi9815** [comment]
Yeah, the config and code snip look correct. 🤔

Could you share me with your full training config file?

**Comment by zzzzzero** [comment]
I also encountered the same problem. During the training process, the validation (val) results were very good, but during inference, the results for the same images were very poor. It seems to be caused by the diffusers version. By downgrading diffusers from 0.32.2 to 0.31.0, the results became normal.

**Comment by zc1023** [comment]
> I also encountered the same problem. During the training process, the validation (val) results were very good, but during inference, the results for the same images were very poor. It seems to be caused by the diffusers version. By downgrading diffusers from 0.32.2 to 0.31.0, the results became normal.

Strangely, my diffusers version is 0.31.0. This problem still occurs.  :D

**Comment by zc1023** [comment]
It works if ` model_config = {
        "union_cond_attn": True,
        "add_cond_attn": False,
        "latent_lora": False,}`.

**Comment by mengxinZHANG** [comment]
> It works if ` model_config = { "union_cond_attn": True, "add_cond_attn": False, "latent_lora": True,}`.

It worked, but I want to know why?

**Comment by mengxinZHANG** [comment]
> Hi [@mengxinZHANG](https://github.com/mengxinZHANG) ,
> 
> It looks like the LoRA controller did not work during trianing.
> 
> Could you please provide the version of PEFE and diffusers?
> 
> Thanks!

peft==0.13.2

**Comment by mengxinZHANG** [comment]
> Hi [@mengxinZHANG](https://github.com/mengxinZHANG) ,
> 
> It looks like the LoRA controller did not work during trianing.
> 
> Could you please provide the version of PEFE and diffusers?
> 
> Thanks!

peft==0.13.2
diffusers==0.32.0

**Comment by zc1023** [comment]
> Hi [@mengxinZHANG](https://github.com/mengxinZHANG) ,
> 
> It looks like the LoRA controller did not work during trianing.
> 
> Could you please provide the version of PEFE and diffusers?
> 
> Thanks!

peft==0.15.1
diffusers==0.31.0

**Comment by potatoker** [comment]
> > It works if ` model_config = { "union_cond_attn": True, "add_cond_attn": False, "latent_lora": True,}`.
> 
> It worked, but I want to know why?

same here, the training callback is good but the inference by hand is bad, Although ` model_config = { "union_cond_attn": True, "add_cond_attn": False, "latent_lora": False,}` in training config, but only ` model_config = { "union_cond_attn": True, "add_cond_attn": False, "latent_lora": True,}` works when inference by hand

**Comment by Yuanshi9815** [comment]
Hi everyone @potatoker @zzzzzero @mengxinZHANG @zc1023 ,

This bug has been fixed. You can pull the latest code.

This bug is caused by #112  which makes the LoRA controller can not work correctly during training.

**Comment by potatoker** [comment]
> Hi everyone [@potatoker](https://github.com/potatoker) [@zzzzzero](https://github.com/zzzzzero) [@mengxinZHANG](https://github.com/mengxinZHANG) [@zc1023](https://github.com/zc1023) ,
> 
> This bug has been fixed. You can pull the latest code.
> 
> This bug is caused by [#112](https://github.com/Yuanshi9815/OminiControl/pull/112) which makes the LoRA controller can not work correctly during training.

after the update, seemed that the inference by hand ok, but the training callback images not generated successfully

**Comment by Yuanshi9815** [comment]
Hi @potatoker ,

I'll check and update it later.

**Comment by Yuanshi9815** [comment]
Hi @potatoker ,

During my testing, the results generated from callback function is looks good. 🤔

**Comment by potatoker** [comment]
> Hi [@potatoker](https://github.com/potatoker) ,
> 
> During my testing, the results generated from callback function is looks good. 🤔

thanks for checking, I will double check it, but this callback will not actually influence the training result right?
